rusqlite_migration/Cargo.toml

```diff
@@ -25,6 +25,7 @@ include_dir = { version = "0.7.3", optional = true }
 tokio = { version = "1.25", features = ["macros"], optional = true }
 tokio-rusqlite = { version = "0.4.0", optional = true }
 log = "0.4"
+take_mut = "0.2.2"
```

Comment:
I’m not thrilled about adding a new dependency here, I thought we could manage with a `Vec<Option<M>>`, once we use a `Vec` in the underlying storage. I’ll take a look.
Also, maybe [std::mem::replace](https://doc.rust-lang.org/std/mem/fn.replace.html) or one of functions linked there could be used instead, I don’t know.

The crate has [29 direct uses](https://lib.rs/crates/take_mut/rev) but seems not be [actively](https://github.com/Sgeo/take_mut/issues/11) [maintenained](https://github.com/Sgeo/take_mut/issues/13) anymore (and the last release is from 2018, and it still uses rust 2015 edition).

I’m fine with small crates that stop evolving once they fill their use case but no new dependency is better if we can avoid it.

Finally, I’m not convinced that this [AssertUnwindSafe](https://github.com/Sgeo/take_mut/blob/master/src/lib.rs#L37) is [actually safe][doc] no matter what closure we give. Interestingly, this [crate](https://github.com/alecmocatta/replace_with/tree/master#replace_with) uses an alternative method to deal with panics. To be clear, I’ve not looked if this library is better on the other criteria to evaluate a dependency.

[doc]: https://doc.rust-lang.org/stable/std/panic/struct.AssertUnwindSafe.html

Reply:
Yep I wasn't happy about adding this dependency. A function like that should have been a part of the standard library. I changed it, still not happy but we can discuss it under my comment there.